backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/IrModuleDeserializer.kt

```diff
@@ -0,0 +1,1130 @@
+            symbol.descriptor, {
+                IrConstructorImpl(
+                    start, end, origin,
+                    it,
+                    Name.identifier(proto.base.name),
+                    deserializeVisibility(proto.base.visibility),
+                    deserializeIrType(proto.base.returnType),
+                    proto.base.isInline,
+                    proto.base.isExternal,
+                    proto.isPrimary
+                )
+
+            })
+
+        deserializeIrFunctionBase(proto.base, constructor as IrFunctionBase, start, end, origin)
+        return constructor
+    }
+
+    private fun deserializeIrField(proto: KonanIr.IrField, start: Int, end: Int, origin: IrDeclarationOrigin): IrField {
+        val symbol = deserializeIrSymbol(proto.symbol) as IrFieldSymbol
+        val field = IrFieldImpl(
+            start, end, origin,
+            symbol,
+            Name.identifier(proto.name),
+            deserializeIrType(proto.type),
+            deserializeVisibility(proto.visibility),
+            proto.isFinal,
+            proto.isExternal,
+            proto.isStatic
+        )
+        val initializer = if (proto.hasInitializer()) deserializeExpression(proto.initializer) else null
+        field.initializer = initializer?.let { IrExpressionBodyImpl(it) }
+
+        return field
+    }
+
+    private fun deserializeModality(modality: KonanIr.ModalityKind) = when (modality) {
+        KonanIr.ModalityKind.OPEN_MODALITY -> Modality.OPEN
+        KonanIr.ModalityKind.SEALED_MODALITY -> Modality.SEALED
+        KonanIr.ModalityKind.FINAL_MODALITY -> Modality.FINAL
+        KonanIr.ModalityKind.ABSTRACT_MODALITY -> Modality.ABSTRACT
+    }
+
+    private fun deserializeIrProperty(
+        proto: KonanIr.IrProperty,
+        start: Int,
+        end: Int,
+        origin: IrDeclarationOrigin
+    ): IrProperty {
+
+        // We start with the field to get its descriptor
+        val backingField = if (proto.hasBackingField()) {
+            deserializeIrField(proto.backingField, start, end, origin)
+        } else null
+
+        val descriptor = backingField?.descriptor
+            ?: if (proto.hasDescriptor()) deserializeDescriptorReference(proto.descriptor) as PropertyDescriptor else null
+                ?: WrappedPropertyDescriptor()
+
+        val property = IrPropertyImpl(
+            start, end, origin,
+            descriptor,
+            Name.identifier(proto.name),
+            deserializeVisibility(proto.visibility),
+            deserializeModality(proto.modality),
+            proto.isVar,
+            proto.isConst,
+            proto.isLateinit,
+            proto.isDelegated,
+            proto.isExternal
+        )
+
+        symbolTable.referenceProperty(descriptor, { property })
+
+        backingField?.descriptor?.let {
+            symbolTable.declareField(UNDEFINED_OFFSET,
```

Comment:
What is it required for?

Reply:
As with other declarations it is used to declare the declaration to the symbolTable. I have moved it to deserializeIrField though.